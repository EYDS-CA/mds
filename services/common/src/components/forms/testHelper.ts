export const inputMeta = {
    touched: false,
    error: false,
    warning: false,
    autofilled: false,
    asyncValidating: false,
    dirty: false,
    dispatch: jest.fn(),
    form: "FORM",
    initial: undefined,
    invalid: false,
    pristine: true,
    submitFailed: false,
    submitting: false,
    valid: true,
    visited: false,
};

export const errorInputMeta = {
    touched: true,
    error: "Error message",
    warning: false,
    autofilled: false,
    asyncValidating: false,
    dirty: true,
    dispatch: jest.fn(),
    form: "FORM",
    initial: undefined,
    invalid: true,
    pristine: true,
    submitFailed: false,
    submitting: false,
    valid: true,
    visited: true,
}

// needs a value and a name
export const inputProps = {
    onBlur: jest.fn(),
    onChange: jest.fn(),
    onDragStart: jest.fn(),
    onDrop: jest.fn(),
    onFocus: jest.fn()
}