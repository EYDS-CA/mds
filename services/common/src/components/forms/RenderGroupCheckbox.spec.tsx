import React from "react";
import { render } from "@testing-library/react";
import RenderGroupCheckbox from "./RenderGroupCheckbox";
import { inputMeta, inputProps } from "./testHelper";

describe("RenderGroupCheckbox", () => {
    it("renders properly", () => {
        const { container } = render(<RenderGroupCheckbox
            id="groupInput"
            label="Mock Label"
            defaultValue={[]}
            options={[
                { label: "Opt 1", value: "opt1" },
                { label: "Opt 2", value: "opt2" },
                { label: "Opt 3", value: "opt3" }
            ]}
            input={{ value: ["opt1", "opt2"], name: "groupInput", ...inputProps }}
            meta={inputMeta}
        />);
        expect(container).toMatchSnapshot();
    })
})