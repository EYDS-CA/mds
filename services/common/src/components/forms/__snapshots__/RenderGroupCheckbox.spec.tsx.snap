// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderGroupCheckbox renders properly 1`] = `
<div>
  <div
    class="ant-form-item ant-form-item-with-help"
  >
    <div
      class="ant-row ant-form-item-row"
    >
      <div
        class="ant-col ant-form-item-label"
      >
        <label
          class=""
          for="groupInput"
          title="Mock Label"
        >
          Mock Label
        </label>
      </div>
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-checkbox-group"
              id="groupInput"
              name="groupInput"
            >
              <label
                class="ant-checkbox-wrapper ant-checkbox-wrapper-checked ant-checkbox-wrapper-in-form-item ant-checkbox-group-item"
              >
                <span
                  class="ant-checkbox ant-checkbox-checked"
                >
                  <input
                    checked=""
                    class="ant-checkbox-input"
                    name="groupInput"
                    type="checkbox"
                    value="opt1"
                  />
                  <span
                    class="ant-checkbox-inner"
                  />
                </span>
                <span>
                  Opt 1
                </span>
              </label>
              <label
                class="ant-checkbox-wrapper ant-checkbox-wrapper-checked ant-checkbox-wrapper-in-form-item ant-checkbox-group-item"
              >
                <span
                  class="ant-checkbox ant-checkbox-checked"
                >
                  <input
                    checked=""
                    class="ant-checkbox-input"
                    name="groupInput"
                    type="checkbox"
                    value="opt2"
                  />
                  <span
                    class="ant-checkbox-inner"
                  />
                </span>
                <span>
                  Opt 2
                </span>
              </label>
              <label
                class="ant-checkbox-wrapper ant-checkbox-wrapper-in-form-item ant-checkbox-group-item"
              >
                <span
                  class="ant-checkbox"
                >
                  <input
                    class="ant-checkbox-input"
                    name="groupInput"
                    type="checkbox"
                    value="opt3"
                  />
                  <span
                    class="ant-checkbox-inner"
                  />
                </span>
                <span>
                  Opt 3
                </span>
              </label>
            </div>
          </div>
        </div>
        <div
          style="display: flex; flex-wrap: nowrap;"
        >
          <div
            class="ant-form-item-explain ant-form-item-explain-connected"
            id="groupInput_help"
            role="alert"
          >
            <div
              class=""
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
